gdx/src/com/badlogic/gdx/math/Intersector.java

```diff
@@ -609,46 +609,111 @@ public static boolean intersectRayBounds (Ray ray, BoundingBox box, Vector3 inte
 	/** Quick check whether the given {@link Ray} and {@link BoundingBox} intersect.
 	 * @return Whether the ray and the bounding box intersect. */
 	static public boolean intersectRayBoundsFast (Ray ray, BoundingBox box) {
-		return intersectRayBoundsFast(ray, box.getCenter(tmp1), box.getDimensions(tmp2));
+		return intersectRayBoundsFastMinMax(ray, box.min, box.max);
 	}
 
 	/** Quick check whether the given {@link Ray} and {@link BoundingBox} intersect.
 	 * @param center The center of the bounding box
 	 * @param dimensions The dimensions (width, height and depth) of the bounding box
 	 * @return Whether the ray and the bounding box intersect. */
 	static public boolean intersectRayBoundsFast (Ray ray, Vector3 center, Vector3 dimensions) {
-		final float divX = 1f / ray.direction.x;
-		final float divY = 1f / ray.direction.y;
-		final float divZ = 1f / ray.direction.z;
-
-		float minx = ((center.x - dimensions.x * 0.5f) - ray.origin.x) * divX;
-		float maxx = ((center.x + dimensions.x * 0.5f) - ray.origin.x) * divX;
-		if (minx > maxx) {
-			final float t = minx;
-			minx = maxx;
-			maxx = t;
-		}
+		// Half dimensions
+		tmp.set(dimensions).scl(0.5f);
+
+		Vector3 min = tmp1.set(center).sub(tmp);
+		Vector3 max = tmp2.set(center).add(tmp);
+		return intersectRayBoundsFastMinMax(ray, min, max);
+	}
+
+	static public boolean intersectRayBoundsFastMinMax (Ray ray, Vector3 min, Vector3 max) {
+		float tmin = 0;
+		float tmax = Float.MAX_VALUE;
+
+		Vector3 p = ray.origin;
+		Vector3 d = ray.direction;
+
+		// Calculate x
+		if (Math.abs(d.x) < MathUtils.FLOAT_ROUNDING_ERROR) {
+			if (p.x < min.x || p.x > max.x) {
+				return false;
+			}
+		} else {
+			float ood = 1f / d.x;
+			float t1 = (min.x - p.x) * ood;
+			float t2 = (max.x - p.x) * ood;
+
+			if (t1 > t2) {
+				float temp = t1;
+				t1 = t2;
+				t2 = temp;
+			}
 
-		float miny = ((center.y - dimensions.y * 0.5f) - ray.origin.y) * divY;
-		float maxy = ((center.y + dimensions.y * 0.5f) - ray.origin.y) * divY;
-		if (miny > maxy) {
-			final float t = miny;
-			miny = maxy;
-			maxy = t;
+			if (t1 > tmin) {
+				tmin = t1;
+			}
+			if (t2 > tmax) {
```

Comment:
how t2 could be greater than tmax which is initialized to max float value?
i think there is a typo for the 3 axis.

Reply:
Whoa, nice catch! The book has a typo.